kornia/contrib/distance_transform.py

```diff
@@ -68,7 +68,7 @@ def distance_transform(
 
         offset: int = i * kernel_size // 2
         out += (offset + cdt) * mask
-        boundary[mask == 1] = 1
+        boundary = torch.where(mask == 1, torch.ones_like(boundary), boundary)

```

Comment:
I would suggest to put torch.ones_like out of the for-loop to reduce the memory creation.

Reply:
I'm not familiar with pytest or gradient check. 
Is it okay if I add a function under `kornia.test.test_contrib.TestConvDistanceTransform`?

```python
class TestConvDistanceTransform:
    def test_loss_grad(self, device, dtype):
        B, C, H, W = 1, 1, 32, 32
        input1 = torch.rand(B, C, H, W, device=device, dtype=dtype, requires_grad=True)
        input2 = torch.rand(B, C, H, W, device=device, dtype=dtype, requires_grad=True)
        tiny_module = nn.Conv2d(1, 1, (3, 3), (1, 1), (1, 1))
        input1 = kornia.contrib.distance_transform(tiny_module(input1))
        input2 = kornia.contrib.distance_transform(input2)
        loss = torch.nn.functional.mse_loss(input1, input2)
        try:
            loss.backward()
        except RuntimeError as e:
            print('Grad check failed for kornia.contrib.distance_transform when calculating regression loss.')
            raise e
```

Reply:
LGTM :)